pkg/cache/admissioncheck.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+type PreemptionPolicy int
+
+const (
+	PreemptAnytime PreemptionPolicy = iota
```

Comment:
Why not api constants?

Reply:
Let's not do such optimisations. The comparison will happen only a handful times during the workload lifecycle.

Reply:
The checks are done for every check of all the workloads pending preemption every time a preemption run starts.